### DISCUSSION TITLE
关于本项目的疑问？

### DISCUSSION THREAD

**Discussion by zqabc** [Django-Simple-API/django-simple-api#36] [General]
我看过fastapi，基于type hint的提示和校验印象深刻，本项目的验证部分类似于他。
我现在一直在用drf，drf也有对应的序列化，可以实现一样的功能，我想问的是：
1. 相比较drf的优点是什么？
2. 如果仅是序列化和校验的话为啥不在drf的基础上做？
3. 类似的用户认证和鉴权等功能在哪里实现？

**Comment by waitan2018** [comment]
这个项目的定位是一个基于 django 的轻量级的插件，而不是一款框架。这个插件只有两个核心功能：自动生成接口文档、自动校验请求参数，以及一些其他的小的工具函数。
至于为什么不基于 drf，那是因为我们认为 drf 太重了，比 django 还重，学习成本太高。且这个项目主要解决的痛点就是“手写接口文档”，所以一定要干净，轻便，不强制用户学习 drf。

**Comment by waitan2018** [comment]
也没有用户认证和鉴权的功能，这只是个解决特定问题的插件。不是个大而全的框架。
